Build on osx arm64

mdpiper opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

mdpiper committed:
Add osx_arm64 platform

mdpiper committed:
Bump build number

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.
I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * No valid build backend found for Python recipe for package `pymt_ecsimplesnow` using `pip`. Python recipes using `pip` need to explicitly specify a build backend in the `host` section. If your recipe has built with only `pip` in the `host` section in the past, you likely should add `setuptools` to the `host` section of your recipe.

mdpiper committed:
Add setuptools to the host requirements

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

mdpiper commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.1, and conda-forge-pinning 2024.10.04.13.32.50

mdpiper committed:
Remove bmi-fortran=1.2 requirement

The package isn't needed because the BMI is included and built locally in ecsimplesnow.

mdpiper committed:
Revert "Remove bmi-fortran=1.2 requirement"

This reverts commit 79e7e81892cd4b1d995cf294695312495396ff52.

Built bmi-fortran=1.2.1 for osx_arm64.

mdpiper commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/11208261736](https://github.com/conda-forge/conda-forge-webservices/actions/runs/11208261736).</sub>

mdpiper committed:
Specify exact version for bmi-fortran

mdpiper committed:
Add cross-compiling build requirements
